src/kv/apply_changes.h

```diff
@@ -70,7 +71,15 @@ namespace kv
     }
 
     bool ok = true;
-    if (has_writes)
+
+    if (expected_rollback_count.has_value())
+    {
+      // Assumes that all maps in the transaction point to the same store
+      auto store = changes.begin()->second.map->get_store();
```

Comment:
Assume we can never get here when `changes` is empty? Else `changes.begin()->second` will go boom.

Reply:
Not in practice, because the only caller that makes use of the expected_rollback_count is the signature Tx, which definitely makes exactly two changes every time. But I agree that making the code defensive and adding a comment is preferable.